Lib/importlib/_bootstrap_external.py

```diff
@@ -91,22 +92,48 @@ def _unpack_uint16(data):
     return int.from_bytes(data, 'little')
 
 
-def _path_join(*path_parts):
-    """Replacement for os.path.join()."""
-    return path_sep.join([part.rstrip(path_separators)
-                          for part in path_parts if part])
+if _MS_WINDOWS:
+    def _path_join(*path_parts):
+        """Replacement for os.path.join()."""
+        if not path_parts:
+            return ""
+        if len(path_parts) == 1:
+            return path_parts[0]
+        root = ""
+        path = []
+        for new_root, tail in map(_os._path_splitroot, path_parts):
+            if new_root.startswith(path_sep_tuple) or new_root.endswith(path_sep_tuple):
+                root = new_root.rstrip(path_separators) or root
+                path = [path_sep + tail]
+            elif new_root.endswith(':'):
+                if root.casefold() != new_root.casefold():
+                    # Drive relative paths have to be resolved by the OS, so we reset the
+                    # tail but do not add a path_sep prefix.
+                    root = new_root
+                    path = [tail]
+                else:
+                    path.append(tail)
+            else:
+                root = new_root or root
+                path.append(tail)
+        if len(path) == 1 and path[0] in path_sep_tuple:
+            # Avoid losing the root's trailing separator when joining with nothing
+            return root + path_sep
```

Comment:
`path` can end up with multiple items where the first item is a backslash, or even multiple slashes, e.g. `_path_join('C:/', '')` and  `_path_join('C:/', 'C:')` both result in ` ['\\', '']`, and `_path_join('C://', 'C:')` results in ` ['\\/', '']`. I had suggested preprocessing it as `path = [p.rstrip(path_separators) for p in path if p]`, which in these cases would result in `['']`. I had an additional check for a UNC path in that case to prevent adding a slash for no good reason, but that's now handledby the initial check `if len(path_parts) == 1`.